framework/machine-learning/inc/MLTensor.h

```diff
@@ -0,0 +1,149 @@
+/* Copyright (c) 2017 SKKU ESLAB, and contributors. All rights reserved.
+ *
+ * Contributor: Gyeonghwan Hong<redcarrottt@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __ML_TENSOR_H__
+#define __ML_TENSOR_H__
+
+#include <iostream>
+#include <string>
+#include <vector>
+#include <map>
+
+namespace MLDataType {
+  enum Value {
+    Char = 1,
+    Int32 = 2,
+    Float = 3,
+    Double = 4
+  };
+
+  inline int size(MLDataType::Value dataType) {
+    switch(dataType) {
+      case MLDataType::Char:
+        return 1;
+        break;
+      case MLDataType::Int32:
+        return 4;
+        break;
+      case MLDataType::Float:
+        return 4;
+        break;
+      case MLDataType::Double:
+        return 8;
+        break;
+      default:
+        return 0;
+        break;
+    }
+  }
+};
+
+class MLTensorLayout {
+  public:
+    MLTensorLayout(int rank, int* shape, MLDataType::Value dataType)
+      : mRank(rank), mDataType(dataType) {
+      this->mShape = new int[rank];
+      for(int i = 0; i < rank; i++) {
+        this->mShape[i] = shape[i];
+      }
+    }
+    ~MLTensorLayout() {
+      delete[] this->mShape;
+    }
+
+    bool isMatched(MLTensorLayout* tensorLayout) {
+      if(tensorLayout->getRank() != this->getRank()) {
+        return false;
+      } else if(tensorLayout->getDataType() != this->getDataType()) {
+        return false;
+      } else {
+        int* thisShape = this->getShape();
+        int* givenShape = tensorLayout->getShape();
+        for(int i = 0; i < this->mRank; i++) {
+          if(thisShape[i] != givenShape[i]) {
+            return false;
+          }
+        }
+        return true;
+      }
+    }
+
+    int getNumElements() {
+      int numElements = 1;
+      for(int i = 0; i < this->mRank; i++) {
+        numElements = numElements * this->mShape[i];
+      }
+      return numElements;
+    }
+
+    int getElementSize() {
+      return MLDataType::size(this->mDataType);
+    }
+
+    int getBytesSize() {
+      return this->getNumElements() * this->getElementSize();
+    }
+
+    int getRank() { return this->mRank; }
+    int* getShape() { return this->mShape; }
+    MLDataType::Value getDataType() { return this->mDataType; }
+    
+  protected:
+    int mRank;
+    int* mShape;
+    MLDataType::Value mDataType;
+};
+
+class MLTensor {
```

Comment:
@sinban04 
```MLTensor``` is defined here, ```framework/machine-learning/src/MLTensor.h```.
Since the body of each member functions are defined in header file, there is no source code file ```MLTensor.cpp```.